### PR TITLE
Activation token in auth controller

### PR DESCRIPTION
**Pull request by mvein**
This is set of changes provided in AuthController in order to support account activation after user registration.
There are some prerequisites before run the new feature:
1) it is necessary to have a proper notification template in order to deal with sending email to registered user. The NotificationTemplate table can be easily fulfilled when seed method is uncommented during first execution (`seed.SeedNotificationTemplates()`) 
2) do not forget to setup MailServiceAuthenticationUserPassword in appsettings.json configuration file, otherwise application will not be able to authenticate to our mailbox. All credentials are stored in  Evernote.

### CODE REVIEW

File: PictureApp.API.Services/EmailNotificationService.cs [hunk 1]
```diff
@@ -1,53 +1,44 @@
...
+            {
+                throw new ArgumentException($"Incorrect format for passed recipient email: `{recipient}`.", e);
+            }
 
-            templateData.GetKeys().ForEach(key => body.Replace(key, templateData.GetValue(key)));
-            return body.ToString();
+            var notification = await _notificationTemplateService.CreateNotification(templateData);
+
+            // Send the email via gateway client
+            await _emailClientProvider.SendAsync(recipientMailAddress, notification.Subject, notification.Body);
```
Comment (anowakowski) [inline PictureApp.API.Services/EmailNotificationService.cs]: one single question to solve my interest ;) send Async functionality is behind 'try and catch' block, its good way? What happens when sendAsync will crash?

File: PictureApp.API/Controllers/AuthController.cs [hunk 2]
```diff
@@ -1,40 +1,68 @@
...
+        [HttpPost("activate")]
+        public async Task<IActionResult> Activate(UserForRegisterActivateDto userForRegisterActivate)
+        {
+            try
+            {
+                await _authService.Activate(userForRegisterActivate.Token);
+            }
+            catch (Exception ex)
+            {
+                if (ex is SecurityTokenExpiredException || ex is EntityNotFoundException)
```
Comment (anowakowski) [inline PictureApp.API/Controllers/AuthController.cs]: is it necessary to Use Only Global Exception on catch block? Maybe we can merge SecurityTokenExpiredException and EntityNotFoundException to main catch block and remove if clausure and throw

File: PictureApp.API.Services/EmailNotificationService.cs (hunk 1 above)
Comment (mvein) [reply PictureApp.API.Services/EmailNotificationService.cs]: In this case 'try and catch' block is only for strict purpose - checking whether passed string represents recipient email is consistent with email format. The MailAddress class does not inlcude TryParse method, hence such approach has been taken. If exception appears on SendAsync than, the global error handler should handle it (I assume).

File: PictureApp.API/Controllers/AuthController.cs (hunk 2 above)
Comment (mvein) [reply PictureApp.API/Controllers/AuthController.cs]: This approach is for catching specific exceptions. First of all, the global one is handled and thereafter Security* and Entiy* are taking into consideration. This was my purpose. What do You mean by merge them?

**Pull request merged**